pymc_marketing/mmm/evaluation.py

```diff
@@ -0,0 +1,535 @@
+
+        Normalization allows for comparison across different data sets and methodologies.
+        NRMSE is one of the key metrics used in Robyn MMMs.
+
+        Parameters
+        ----------
+        y_true : np.ndarray
+            True values for target metric
+        y_pred : np.ndarray
+            Predicted values for target metric
+
+        Returns
+        -------
+        float
+            Normalized root mean square error.
+        """
+        return root_mean_squared_error(y_true, y_pred) / (y_true.max() - y_true.min())
+
+    @staticmethod
+    def nmae(y_true: np.ndarray, y_pred: np.ndarray) -> float:
+        """Calculate the Normalized Mean Absolute Error (NMAE).
+
+        Normalization allows for comparison across different data sets and methodologies.
+
+        Parameters
+        ----------
+        y_true : np.ndarray
+            True values for target metric
+        y_pred : np.ndarray
+            Predicted values for target metric
+
+        Returns
+        -------
+        float
+            Normalized mean absolute error.
+        """
+        return mean_absolute_error(y_true, y_pred) / (y_true.max() - y_true.min())
+
+    def plot_hdi_forest(
+        self,
+        var_names: list,
+        hdi_prob: float = 0.94,
+        figsize: tuple = (12, 8),
+        **plot_kwargs,
+    ) -> plt.Figure:
+        """Plot a forest plot to compare high-density intervals (HDIs).
+
+        Plot a forest plot to compare high-density intervals (HDIs) from a given set of
+        posterior distributions, as well as their r-hat statistics.
+
+        Parameters
+        ----------
+        var_names : list
+            List of variable names to include in the forest plot.
+        hdi_prob : float, optional
+            The probability mass of the highest density interval. Defaults to 0.94.
+        figsize : tuple, optional
+            Figure size in inches. Defaults to (12, 8).
+        **plot_kwargs
+            Additional keyword arguments to pass to the az.plot_forest method.
+
+        Returns
+        -------
+        plt.Figure
+            The matplotlib figure object.
+
+        Raises
+        ------
+        ValueError
+            If the required attributes (idata) are not found.
+        """
+        if not hasattr(self.model, "idata") or self.model.idata is None:
+            raise ValueError(
+                "The model object must have a non-None 'idata' attribute. "
+                "Ensure you've called model.fit() before running diagnostics."
+            )
+
+        fig, ax = plt.subplots(figsize=figsize, ncols=2)
+        az.plot_forest(
+            data=self.model.idata,
+            var_names=var_names,
+            combined=True,
+            ax=ax,
+            hdi_prob=hdi_prob,
+            r_hat=True,
+            **plot_kwargs,
+        )
+
+        fig.suptitle(f"Posterior Distributions: {hdi_prob:.1%} HDI")
+
+        return fig
+
+    def plot_prior_vs_posterior(
```

Comment:
Might be better suited in the MMM class itself. Heavy coupling using the channel_columns attributes, etc

Reply:
yes! I agree! this does not need to be part of an "evaluator"